Fixed same-name imports from wrong package

EthanKim8683 opened the pull request:
<!-- Thank you for your contribution! -->
## Changes

Fixes `add_import()` logic that incorrectly imports something from `sqlalchemy` if `sqlalchemy` has it. So, `UUID`, for example, is supposed to be imported from `uuid`, but `sqlalchemy` exports `UUID` as well, resulting in the original code importing from `sqlalchemy` instead.

Closes #415

<!-- Please give a short brief about these changes. -->

## Checklist

If this is a user-facing code change, like a bugfix or a new feature, please ensure that
you've fulfilled the following conditions (where applicable):

- [x] You've added tests (in `tests/`) which would fail without your patch
- [x] You've added a new changelog entry (in `CHANGES.rst`).

If this is a trivial change, like a typo fix or a code reformatting, then you can ignore
these instructions.

### Updating the changelog

If there are no entries after the last release, use `**UNRELEASED**` as the version.
If, say, your patch fixes issue <span>#</span>123, the entry should look like this:

```
- Fix big bad boo-boo in task groups
  (`#123 <https://github.com/agronholm/sqlacodegen/issues/123>`_; PR by @yourgithubaccount)
```

If there's no issue linked, just link to your pull request instead by updating the
changelog after you've created the PR.

agronholm commented:
I at least would like to understand why this happens before we merge and move on.

agronholm commented:
First of all, the import grouping is wrong. It should group `uuid` higher as it's from the standard library. Second, why is it not doing `from uuid import UUID`?

EthanKim8683 commented:
Ah, the reason why the declarative generator's output doesn't have this bug is because of its `render_python_type()` method, which adds the _module_ import for the type, and then uses the type like `f"{module}.{name}"` which avoids the mix-up.

EthanKim8683 commented:
@sheinbergon I agree. It fixes naming collisions between imports too, which I've seen become a problem if you import both `uuid.UUID` and `sqlalchemy.UUID` as `UUID`.

sheinbergon commented:
Great. Would you be able to fix your PR accommodate that change?

EthanKim8683 commented:
Hey guys, sorry about the wait. I've made the changes we talked about, plus a couple more. All in all, here's the list of changes:
- `add_import` checks whether imports belong to the `sqlalchemy` package before attempting to import from `sqlalchemy` (from before)
- Added `SQLModelGenerator` test for `uuid.UUID` (from before)
- `render_column_attribute` now consistently imports entire Python packages instead of individual imports from packages (as discussed)
- `group_imports` now groups module imports as well (as discussed, before it was just packages)
- `builtin_module_names` has been renamed to `stdlib_module_names` (based on the way it was used in `group_imports`, I figured it'd be more fitting)
- `stdlib_module_names` contains`sys.stdlib_module_names` if running Python 3.10 or higher, or gets the stdlib module names from `stdlib-list` otherwise.
- Fixed one `DataclassGenerator` test which didn't group module names correctly
-  Auto-incrementing primary keys are no longer optional in `SQLModelGenerator`, consistent with the other generators
- Updated `SQLModelGenerator` tests to reflect change above

EthanKim8683 commented:
Happy to help :)

sheinbergon commented:
@agronholm can you please share your thoughts. I'd like to include this PR in the upcoming 3.1 release

agronholm commented:
> @EthanKim8683 almost forgot. Can you please add a short description to `CHANGES.rst`?

Yeah, it looks like they checked the box for an added changelog entry, without actually having done so.

sheinbergon commented:
@EthanKim8683  can we please follow up on 2 additional minor fixes so we can have this merged? I'd like this to be included in the upcoming release

sheinbergon commented:
@EthanKim8683 a kind reminder
